lib/wickedbase.pm

```diff
@@ -737,6 +738,51 @@ sub record_console_test_result {
     $self->write_resultfile($filename, $content);
 }
 
+sub check_logs {
+    my $self = shift;
+    my @units = qw(wickedd-nanny wickedd-dhcp4 wickedd-dhcp6 wicked wickedd);
+    my $default_exclude = 'wickedd=process \d+ has not exited yet; now doing a blocking waitpid';
+    $default_exclude .= ',wickedd-dhcp6=Link-local IPv6 address is marked duplicate:';
+    $default_exclude .= ',wickedd-nanny=: device has been deleted';
+    $default_exclude .= ',wickedd-dhcp4=unable to confirm lease';
+    $default_exclude .= ',wickedd-nanny=: call to org.opensuse.Network.Interface.waitLinkUp\(\) failed: General failure';
+    $default_exclude .= ',wickedd-nanny=: call to org.opensuse.Network.Interface.waitLinkUp\(\) failed: Object does not support requested method';
+    $default_exclude .= ',wickedd-nanny=: failed to bring up device, still continuing';
+    $default_exclude .= ',wickedd=error retrieving tap attribute from sysfs';
+
+    my $exclude_var = get_var(WICKED_CHECK_LOG_EXCLUDE => $default_exclude);
+    my $exclude_test_var = get_var('WICKED_CHECK_LOG_EXCLUDE_' . $self->{name}, '');
+
+    my @excludes = split(/(?<!\\),/, "$exclude_var,$exclude_test_var");
+    @excludes = map { my $v = trim($_); length($v) > 0 ? $v : () } @excludes;
+
+    for my $unit (@units) {
+        my $cmd = "journalctl -q -p 3 -x -u $unit";
+        for my $exclude (@excludes) {
+            my ($unit_match, $regex) = split(/\s*=\s*/, $exclude, 2);
+            if ($unit_match =~ /^all$/i || $unit_match eq $unit) {
+                $cmd .= "\\\n    | grep -vP '$regex'";
+            }
+        }
+        my $out = trim(script_output($cmd, proceed_on_failure => 1));
+        if (length($out) > 0) {
+            my $msg = "$cmd\n\n$out\n\n";
+            $msg .= "Use WICKED_CHECK_LOG_EXCLUDE to change filter!\n";
+            $msg .= "  WICKED_CHECK_LOG_EXCLUDE=$exclude_var\n";
+            $msg .= '  WICKED_CHECK_LOG_EXCLUDE_' . $self->{name} . "=$exclude_test_var\n";
+            $msg .= "Control if test fail with WICKED_CHECK_LOG_FAIL default off.\n";
+            record_info('LOG-ERROR', $msg, result => 'fail');
+            $self->result('fail') if get_var(WICKED_CHECK_LOG_FAIL => 0) && $self->{name} ne 'before_test';
+        }
+    }
+}
+
+sub reboot {
+    my ($self) = @_;
+    $self->check_logs();
+    serial_terminal::reboot();
```

Comment:
I think it make sense to do this check also **after** reboot ( but keeping before reboot check too )

Reply:
ok